controllers/registry.go

```diff
@@ -243,33 +243,31 @@ func replaceCarriageReturn(data map[string][]byte, logger logr.Logger) map[strin
 	return data
 }
 
-func (r *DPAReconciler) getSecretNameAndKeyforBackupLocation(bslspec oadpv1alpha1.BackupLocation) (string, string) {
-
-	if bslspec.CloudStorage != nil {
-		if bslspec.CloudStorage.Credential != nil {
-			return bslspec.CloudStorage.Credential.Name, bslspec.CloudStorage.Credential.Key
+func (r *DPAReconciler) getSecretNameAndKeyFromCloudStorage(cloudStorage *oadpv1alpha1.CloudStorageLocation) (string, string, error) {
+	if cloudStorage.Credential != nil {
+		secretName, secretKey := cloudStorage.Credential.Name, cloudStorage.Credential.Key
+		err := r.verifySecretContent(secretName, secretKey)
+		if err != nil {
```

Comment:
be careful in these situations: note that the if is not necessary, i.e. this
```go
		if err != nil {
			return secretName, secretKey, err
		}

		return secretName, secretKey, err
```
is the same as 
```go
		return secretName, secretKey, err
```

Reply:
ah right! we need to return in both cases. updating this